test/basic.test.js

```diff
@@ -0,0 +1,18 @@
+import { test } from 'tap'
+import Fastify from 'fastify'
+
+import slowDownPlugin from '../index.js'
+import { HEADERS, DEFAULT_OPTIONS } from '../lib/constants.js'
+
+test('Should work as a normal API', async t => {
+  const fastify = Fastify()
+  await fastify.register(slowDownPlugin)
+  t.teardown(() => fastify.close())
+
+  fastify.get('/', async () => 'Hello fastify-slow-down!')
+
+  const res = await fastify.inject('/')
+
+  t.equal(res.statusCode, 200)
+  t.equal(res.headers[HEADERS.limit], DEFAULT_OPTIONS.delayAfter)
```

Comment:
check the response as well. hence why it makes more sense to use a simpler response string or object

Reply:
you are never testing the remaining header. it should be tested both in the simple case when we're not delaying requests and in tests where we're delaying requests